wandb/sdk_py27/internal/tb_watcher.py

```diff
@@ -215,8 +215,14 @@ def _thread_body(self):
             try:
                 for event in self._generator.Load():
                     self.process_event(event)
-            except self.directory_watcher.DirectoryDeletedError:
-                break
+            except (
+                self.directory_watcher.DirectoryDeletedError,
+                StopIteration,
+                RuntimeError,
+            ) as e:
+                # When listing s3 the directory may not yet exist, or could be empty
+                logger.debug("Encountered tensorboard directory watcher error: %s", e)
+                continue
```

Comment:
This might end up in a tight loop, maybe a time.sleep(1) here is a good idea? or restructuring the loop to make sure the existing time.sleep(1) is executed in this path.
Maybe process_event makes sure this doesnt happen, but it seems good to be safe in an error path.

Reply:
https://github.com/wandb/client/pull/1722 will fix the unittest failure if it gets an approval. :)

Reply:
Yeah, I was thinking the same thing.  I'll add a 5 second sleep.